### PR TITLE
fix #21

### LINKED ISSUE #21: plugin fails for iOS on 204
Hello, when I upload data to amazon bucket, it returns 204 and the plugin on iOS fails
hasResponseError = response!.statusCode != 200 && response!.statusCode != 201
the issue is in this line, could it check in a better way the status code?

### PR DESCRIPTION
**Pull request by onikiri2007**
This will fix to handle other success status code based on HTTP spec

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,7 @@
+## 1.0.5+2
+
+- fix #21 - handle other successful status code (from http spec) in iOS
```
Comment (ened) [inline CHANGELOG.md]: Could you also write that extra check for android or shall I?

File: android/src/main/java/com/bluechilli/flutteruploader/UploadWorker.java [hunk 2]
```diff
@@ -129,8 +129,10 @@ public Result doWork() {
 
         if (file.exists() && file.isFile()) {
           fileExistsCount++;
+          String mimeType = GetMimeType(item.getPath());
+          MediaType contentType = MediaType.parse(mimeType);
           RequestBody fileBody =
-              RequestBody.create(MediaType.parse(GetMimeType(item.getPath())), file);
+              RequestBody.create(file, contentType);
```
Comment (ened) [inline android/src/main/java/com/bluechilli/flutteruploader/UploadWorker.java]: This seems like it flips the actual parameter?

**Comment by onikiri2007** [comment]
I flipped because  other way seems depricated.

**Comment by ened** [comment]
OK, I did not check the source code warning in any IDE, just looked like a surprising thing. All good, then I think.

File: CHANGELOG.md (hunk 1 above)
Comment (onikiri2007) [reply CHANGELOG.md]: android already does the check or rather the okHttp itself does it with isSuccessful method

**Comment by onikiri2007** [comment]
> Hi @onikiri2007 - I'm not sure about the Android changes and would suggest to only fix the iOS status code issue #21 in this PR.
> I will prepare a OKHTTP update in the coming week (to bump it to latest OkHttp 4.x.x) and review the code at the same time.

yap it's reversed

**Comment by ened** [comment]
@onikiri2007 please rebase once again (formatting in Android part gone wrong), then we are ready.

**Comment by onikiri2007** [comment]
rebased

**Pull request merged**